tests/test__minhash.py

```diff
@@ -1041,6 +1041,73 @@ def test_abundance_simple():
     assert a.get_mins(with_abundance=True) == {2110480117637990133: 2}
 
 
+def test_add_hash_with_abundance():
+    a = MinHash(20, 5, False, track_abundance=True)
+
+    a.add_hash_with_abundance(10, 1)
+    assert a.get_mins(with_abundance=True) == {10: 1}
+
+    a.add_hash_with_abundance(20, 2)
+    assert a.get_mins(with_abundance=True) == {10: 1, 20: 2}
+
+    a.add_hash_with_abundance(10, 2)
+    assert a.get_mins(with_abundance=True) == {10: 3, 20: 2}
+
+
+def test_set_abundance_clear_on_empty_minhash():
+    a = MinHash(20, 5, False, track_abundance=True)
+    b = MinHash(20, 5, False, track_abundance=True)
+
+    a.set_abundances({1: 3, 2: 4}, clear=True)
+    b.set_abundances({1: 3, 2: 4}, clear=False)
+
+    assert a.get_mins() == b.get_mins()
+
+
+def test_set_abundance_clear_default():
+    a = MinHash(20, 5, False, track_abundance=True)
+
+    a.add_hash(10)
+    assert a.get_mins() == [10]
+    assert a.get_mins(with_abundance=True) == {10: 1}
+
+    a.set_abundances({20: 2})
+    assert a.get_mins(with_abundance=True) == {20: 2}
+
+
+def test_set_abundance_no_clear():
+    a = MinHash(20, 5, False, track_abundance=True)
+
+    a.add_hash(10)
+    assert a.get_mins(with_abundance=True) == {10: 1}
+    
+    a.set_abundances({20: 1, 30: 4}, clear=False)
+    assert a.get_mins(with_abundance=True) == {10: 1, 20: 1, 30: 4}
+
+
+def test_set_abundance_no_clear_with_set_hash():
+    a = MinHash(20, 5, False, track_abundance=True)
+
+    a.add_hash(10)
+    assert a.get_mins(with_abundance=True) == {10: 1}
+    
+    a.set_abundances({10: 3}, clear=False)
+    assert a.get_mins(with_abundance=True) == {10: 3}
+
+    a.set_abundances({20: 5, 10: 2}, clear=False)
+    assert a.get_mins(with_abundance=True) == {10: 2, 20: 5}
```

Comment:
so I think this assert is maybe getting the wrong result - we want `set_abundances(..., clear=False)` to add the abundances, so here the abundance of 10 would be 5 instead of 2.

note that dictionary updates using the dictionary unpacking mechanism don't sum the values  for colliding keys (as you would expect, because dictionaries store many types of values, and there's no reason to special case int in this way).
```
>>> x = { 1: 1, 2: 1 }
>>> y = { 1: 1, 3: 1 }
>>> {**x, **y}
{1: 1, 2: 1, 3: 1}
```

Reply:
Wait so would it be better for `set_abundances(..., clear=False)` to add the abundances or to override and set new ones? Because I think I'm getting opposing opinions from you and @luizirber. Also, If it adds the abundances, wouldn't it have the exact same functionality as `add_hash_with_abundance(...)`?

Reply:
Great discussion! I'm putting this in a fuller comment now, but `add_hash_with_abundance(...)` takes one hash at a time, while `set_abundances(...)` takes multiple hashes and is more efficient for large MinHashes.